From 1d32fab675e2a9b6fb81f10245608a0c746839cc Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:06:19 -0400
Subject: [usb] uas: Pack iu struct definitions

Message-id: <1410887463-20674-308-git-send-email-dzickus@redhat.com>
Patchwork-id: 95322
O-Subject: [RHEL7 PATCH 307/591] uas: Pack iu struct definitions
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit d24354bbf78f9194ec21087130a69b84864e50df
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Mon Oct 21 11:15:11 2013 +0100

    uas: Pack iu struct definitions

    The iu struct definitions are usb packet definitions, so no alignment should
    happen. Notice that assuming 32 bit alignment this does not make any
    difference at all.

    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/usb/uas.h b/include/linux/usb/uas.h
index 772b66b..3fc8e8b 100644
--- a/include/linux/usb/uas.h
+++ b/include/linux/usb/uas.h
@@ -9,7 +9,7 @@ struct iu {
  __u8 iu_id;
  __u8 rsvd1;
  __be16 tag;
-};
+} __attribute__((__packed__));
 
 enum {
  IU_ID_COMMAND  = 0x01,
@@ -52,7 +52,7 @@ struct command_iu {
  __u8 rsvd7;
  struct scsi_lun lun;
  __u8 cdb[16]; /* XXX: Overflow-checking tools may misunderstand */
-};
+} __attribute__((__packed__));
 
 struct task_mgmt_iu {
  __u8 iu_id;
@@ -62,7 +62,7 @@ struct task_mgmt_iu {
  __u8 rsvd2;
  __be16 task_tag;
  struct scsi_lun lun;
-};
+} __attribute__((__packed__));
 
 /*
  * Also used for the Read Ready and Write Ready IUs since they have the
@@ -77,7 +77,7 @@ struct sense_iu {
  __u8 rsvd7[7];
  __be16 len;
  __u8 sense[SCSI_SENSE_BUFFERSIZE];
-};
+} __attribute__((__packed__));
 
 struct response_iu {
  __u8 iu_id;
@@ -85,7 +85,7 @@ struct response_iu {
  __be16 tag;
  __u8 add_response_info[3];
  __u8 response_code;
-};
+} __attribute__((__packed__));
 
 struct usb_pipe_usage_descriptor {
  __u8  bLength;
-- 
1.7.1